fair/tools/unifiedtools.py

```diff
@@ -229,13 +312,15 @@ def return_np_function_arg_list(inp_df, cfg, concentration_mode=False):
 
     ext_forcing_np = convert_df_to_numpy(cfg["ext_forcing"])[0]
 
-    inp_df = inp_df.iloc[inp_df.index.astype("str").str.lower().argsort()]
-    time_index = inp_df.index
+    time_index = inp_df.timeseries().columns
```

Comment:
You might want to check you've got integers here. Some pyam dataframes have funky time axes so a quick assert is worth it

Reply:
Hang on, what type is `inp_df`? If it's an `ScmRun`, you can just do `inp_df.timeseries(time_axis="year")` and then you'll get years as timesteps rather than pandas timestamps